storage/src/main/java/org/apache/uniffle/storage/factory/ShuffleHandlerFactory.java

```diff
@@ -119,13 +119,11 @@ public ClientReadHandler createSingleReplicaClientReadHandler(CreateShuffleReadH
   private ClientReadHandler getMemoryClientReadHandler(CreateShuffleReadHandlerRequest request, ShuffleServerInfo ssi) {
     ShuffleServerClient shuffleServerClient = ShuffleServerClientFactory.getInstance().getShuffleServerClient(
         ClientType.GRPC.name(), ssi);
-    Roaring64NavigableMap expectTaskIds;
-    if (request.isExpectedTaskIdsBitmapFilterEnable()) {
+    Roaring64NavigableMap expectTaskIds = null;
+    if (request.isExpectedTaskIdsBitmapFilterEnable() || request.getShuffleServerInfoList().size() > 1) {
```

Comment:
Could we put the similar logic in the same place? It's more easy to manage. 
https://github.com/apache/incubator-uniffle/blob/66c752fe3ecd6d4979d3539fe22265d4239156a9/client-spark/spark3/src/main/java/org/apache/spark/shuffle/reader/RssShuffleReader.java#L125